### PR TITLE
GSdx: Add option to Disable Safe Features

### PR DESCRIPTION
**Pull request by lightningterror**
Idea is to disable multiple features in 1 option to avoid gui clutter.

Unscale Point and Line:
This partially reverts PR #2710 restoring Unscale Point and Line but as a disable option.
It can help Xenosaga games.

GS Memory clearning:
This disables GS memory clearing to be done on the CPU, by default both gpu and cpu do memory clearing
which is more accurate.
It can help Kingdom Hearts games.

### CODE REVIEW

File: plugins/GSdx/Renderers/HW/GSRendererHW.cpp [hunk 1]
```diff
@@ -41,18 +41,20 @@ GSRendererHW::GSRendererHW(GSTextureCache* tc)
...
 		m_userhacks_align_sprite_X       = theApp.GetConfigB("UserHacks_align_sprite_X");
 		m_userhacks_round_sprite_offset  = theApp.GetConfigI("UserHacks_round_sprite_offset");
-		m_userhacks_disable_gs_mem_clear = theApp.GetConfigB("UserHacks_DisableGsMemClear");
 		m_userHacks_HPO                  = theApp.GetConfigI("UserHacks_HalfPixelOffset");
 		m_userHacks_merge_sprite         = theApp.GetConfigB("UserHacks_merge_pp_sprite");
 		m_userhacks_tcoffset_x           = theApp.GetConfigI("UserHacks_TCOffsetX") / -1000.0f;
 		m_userhacks_tcoffset_y           = theApp.GetConfigI("UserHacks_TCOffsetY") / -1000.0f;
 		m_userhacks_tcoffset             = m_userhacks_tcoffset_x < 0.0f || m_userhacks_tcoffset_y < 0.0f;
 	} else {
+		m_userhacks_disable_gs_mem_clear = true;
```
Comment (willkuer) [inline plugins/GSdx/Renderers/HW/GSRendererHW.cpp]: Switching this from false to true will affect old inis

File: plugins/GSdx/Renderers/HW/GSRendererHW.h [hunk 2]
```diff
@@ -145,6 +145,7 @@ class GSRendererHW : public GSRenderer
 
 	int m_userhacks_round_sprite_offset;
 	int m_userHacks_HPO;
+	bool m_userHacks_disable_unscale_ptln;
```
[the remark this answers is not in the document]
Comment (lightningterror) [reply plugins/GSdx/Renderers/HW/GSRendererHW.h]: Ah yes I didn't notice the naming, I was probably planning to do it differently but didn't and forgot to change the name.

File: plugins/GSdx/Renderers/HW/GSRendererHW.cpp [hunk 3]
```diff
@@ -41,18 +41,20 @@ GSRendererHW::GSRendererHW(GSTextureCache* tc)
 	m_large_framebuffer  = theApp.GetConfigB("large_framebuffer");
 	m_accurate_date = theApp.GetConfigI("accurate_date");
 	if (theApp.GetConfigB("UserHacks")) {
+		m_userhacks_disable_gs_mem_clear = !theApp.GetConfigB("UserHacks_Disable_GsMemClear_and_UnscalePL");
```
[the remark this answers is not in the document]
Comment (lightningterror) [reply plugins/GSdx/Renderers/HW/GSRendererHW.cpp]: Actually I changed the behavior at https://github.com/PCSX2/pcsx2/pull/2794/files#diff-52dbe32eef0f2270ea51de3fdcaec5aaR994

**Comment by turtleli** [comment]
So why are the options being combined?

**Comment by lightningterror** [comment]
@gregory38 suggested it, since the options are used in rare cases.

**Comment by gregory38** [comment]
Yes my request. I know that the 2 hacks are totally different. But a generic" disable safe hack" would be nice to avoid cluttering the GUI. Discussion is open. And a better option name would be nice too.

**Comment by lightningterror** [comment]
Is anyone fine with it so I can merge this ?

**Comment by lightningterror** [comment]
I'll go ahead and merge this. Tooltip and name have already been updated.

**Pull request merged**